vllm/model_executor/guided_decoding.py

```diff
@@ -0,0 +1,36 @@
+from enum import Enum
+from typing import Union
+from types import SimpleNamespace
+from pydantic import BaseModel
+try:
+    from outlines.serve.vllm import JSONLogitsProcessor, RegexLogitsProcessor
+except ImportError as e:
+    raise ValueError(
+        "Please install 'outlines' (pip install outlines) to use guided generation."
+    ) from e
+
+
+class GuidedDecodingMode(Enum):
+    JSON = "json"
+    REGEX = "regex"
+    # TODO: add grammar, choice
+
+
+def get_guided_decoding_logits_processor(guided_spec: Union[str, dict, BaseModel], mode: GuidedDecodingMode, tokenizer):
+    def dummy_llm():
```

Comment:
Maybe we should be improving this on our end and making the logits processor accept a tokenizer as an argument. This would require users to pass `llm.engine.tokenizer.tokenizer` though.

Additionally, `JSONLogitsProcessor` and `RegexLogitsProcessor` aren't currently compatible with vllms `apply_logits_processors` without a patch. However this will be resolved when https://github.com/outlines-dev/outlines/pull/628 is merged.